featuretools/demo/flight.py

```diff
@@ -1,68 +1,219 @@
+
+    # Combine strings like 0130 (1:30 AM) with dates (2017-01-01)
+    clean_data['scheduled_dep_time'] = clean_data['scheduled_dep_time'].apply(lambda x: str(x)) + clean_data['flight_date'].astype(str)
+
+    # Parse combined string as a date
+    clean_data.loc[:, 'scheduled_dep_time'] = pd.to_datetime(
+        clean_data['scheduled_dep_time'], format='%H%M%Y-%m-%d', errors='coerce')
+
+    clean_data['scheduled_elapsed_time'] = pd.to_timedelta(clean_data['scheduled_elapsed_time'], unit='m')
+
+    clean_data = _reconstruct_times(clean_data)
+
+    # Create a time index 6 months before scheduled_dep
+    clean_data.loc[:, 'time_index'] = clean_data['scheduled_dep_time'] - \
+        pd.Timedelta('120d')
+
+    # A null entry for a delay means no delay
+    clean_data = _fill_labels(clean_data)
+
+    # Nulls for scheduled values are too problematic. Remove them.
+    clean_data = clean_data.dropna(
+        axis='rows', subset=['scheduled_dep_time', 'scheduled_arr_time'])
+
+    # Make a flight id. Define a flight as a combination of:
+    # 1. carrier 2. flight number 3. origin airport 4. dest airport
+    clean_data.loc[:, 'flight_id'] = clean_data['carrier'] + '-' + \
+        clean_data['flight_num'].apply(lambda x: str(x)) + ':' + clean_data['origin'] + '->' + clean_data['dest']
+
+    return clean_data
+
+
+def _fill_labels(clean_data):
+    labely_columns = ['arr_delay', 'dep_delay', 'carrier_delay', 'weather_delay',
+                      'national_airspace_delay', 'security_delay',
+                      'late_aircraft_delay', 'cancelled', 'diverted',
+                      'taxi_in', 'taxi_out', 'air_time']
+    for col in labely_columns:
+        clean_data.loc[:, col] = clean_data[col].fillna(0)
+
+    return clean_data
+
+
+def _reconstruct_times(clean_data):
+    """ Reconstruct departure_time, scheduled_dep_time,
+        arrival_time and scheduled_arr_time by adding known delays
+        to known times. We do:
+            - dep_time is scheduled_dep + dep_delay
+            - arr_time is dep_time + taxiing and air_time
+            - scheduled arrival is scheduled_dep + scheduled_elapsed
+    """
+    clean_data.loc[:, 'dep_time'] = clean_data['scheduled_dep_time'] + \
+        pd.to_timedelta(clean_data['dep_delay'], unit='m')
+
+    clean_data.loc[:, 'arr_time'] = clean_data['dep_time'] + \
+        pd.to_timedelta(clean_data['taxi_out'] +
+                        clean_data['air_time'] +
+                        clean_data['taxi_in'], unit='m')
+
+    clean_data.loc[:, 'scheduled_arr_time'] = clean_data['scheduled_dep_time'] + \
+        clean_data['scheduled_elapsed_time']
+    return clean_data
+
+
+def filter_data(clean_data,
+                month_filter=None,
+                categorical_filter=None):
+    if month_filter is not None:
+        tmp = False
+        for month in month_filter:
+            tmp = tmp | (clean_data['scheduled_dep_time'].apply(lambda x: x.month) == month)
```

Comment:
can this be `clean_data['scheduled_dep_time'].dt.month == month`? if that works, it's probably much faster than apply

Reply:
also, you should try to use an isin here rather than iterate over each month